tests/DeepCopyTest/DeepCopyTest.php

```diff
@@ -42,6 +42,17 @@ public function testPropertyObjectCopy()
         $this->assertDeepCopyOf($o, $deepCopy->copy($o));
     }
 
+    public function testPropertyObjectCopyWithDateTimes()
+    {
+        $o = new A();
+        $o->date1 = new \DateTime();
+        $o->date2 = new \DateTimeImmutable();
+
+        $deepCopy = new DeepCopy();
+
+        $this->assertDeepCopyOf($o, $deepCopy->copy($o));
```

Comment:
Write a test that mutates `$o->date1` and assert that the data is different in the deep copy.